pkg/cmd/roachtest/tests/validate_system_schema_after_version_upgrade.go

```diff
@@ -68,67 +66,46 @@ func registerValidateSystemSchemaAfterVersionUpgrade(r registry.Registry) {
-					*output = obtainSystemSchema(ctx, t, u, node)
-				}
-			}
-
-			// Wipe nodes in this test's cluster.
-			wipeClusterStep := func(nodes option.NodeListOption) versionStep {
-				return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-					u.c.Wipe(ctx, false /* preserveCerts */, nodes)
-				}
-			}
-
 			// Compare whether two strings are equal -- used to compare expected and actual.
-			validateEquivalenceStep := func(str1, str2 *string) versionStep {
-				return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-					if *str1 != *str2 {
-						diff, diffErr := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
-							A:       difflib.SplitLines(*str1),
-							B:       difflib.SplitLines(*str2),
-							Context: 5,
-						})
-						if diffErr != nil {
-							diff = diffErr.Error()
-							t.Errorf("failed to produce diff: %v", diffErr)
-						}
-						t.Fatalf("After upgrading, `USE system; SHOW CREATE ALL TABLES;` "+
-							"does not match expected output after version upgrade."+
-							"\nDiff:\n%s", diff)
+			validateEquivalence := func(str1, str2 string) error {
+				if str1 != str2 {
+					diff, diffErr := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
+						A:       difflib.SplitLines(str1),
+						B:       difflib.SplitLines(str2),
+						Context: 5,
+					})
+					if diffErr != nil {
+						return errors.Newf("failed to produce diff: %v", diffErr)
 					}
-					t.L().Printf("validating succeeded:\n%v", *str1)
+					return errors.Newf("After upgrading, `USE system; SHOW CREATE ALL TABLES;` "+
+						"does not match expected output after version upgrade."+
+						"\nDiff:\n%s", diff)
 				}
+				return nil
 			}
 
-			u := newVersionUpgradeTest(c,
-				// Start the node with the latest binary version.
-				uploadAndStart(c.Node(1), clusterupgrade.CurrentVersion()),
-
-				// Obtain expected output from the node.
-				obtainSystemSchemaStep(1, &expected),
-
-				// Wipe the node.
-				wipeClusterStep(c.Node(1)),
-
-				// Restart the node with a previous binary version.
-				uploadAndStart(c.Node(1), predecessorVersion),
-
-				// Upgrade the node version.
-				binaryUpgradeStep(c.Node(1), clusterupgrade.CurrentVersion()),
-
-				// Wait for the cluster version to also bump up to make sure the migration logic is run.
-				waitForUpgradeStep(c.Node(1)),
-
-				// Obtain the actual output on the upgraded version.
-				obtainSystemSchemaStep(1, &actual),
-
-				// Compare the results.
-				validateEquivalenceStep(&expected, &actual),
+			binary := uploadVersion(ctx, t, c, c.All(), clusterupgrade.CurrentVersion())
+			if err := clusterupgrade.StartWithSettings(
+				ctx, t.L(), c, c.All(), option.DefaultStartOpts(), install.BinaryOption(binary),
+			); err != nil {
+				t.Fatal(err)
+			}
+			expected = obtainSystemSchema(ctx, t.L(), c, 1)
+			c.Wipe(ctx, false /* preserveCerts */, c.All())
+
+			mvt := mixedversion.NewTest(ctx, t, t.L(), c, c.All(), mixedversion.NeverUseFixtures)
```

Comment:
> Is this test incompatible with fixtures?

According to this comment, yes.
```
// NeverUseFixtures is an option that can be passed to `NewTest` to
// disable the use of fixtures in the test. Necessary if the test
// wants to use a number of cockroach nodes other than 4.
```

> I believe you'll want mixedversion.NumUpgrades(1) here to force we are only performing one upgrade in this test. Otherwise it might choose to perform several upgrades in the same test: it go through 22.2 -> 23.1 -> master. AfterUpgradeFinalized is called after each upgrade.

Ah nice, thanks for the pointer.